common/src/main/java/com/bakdata/quick/common/api/client/mirror/MirrorHost.java

```diff
@@ -14,37 +14,54 @@
  *    limitations under the License.
  */
 
-package com.bakdata.quick.common.api.model.mirror;
+package com.bakdata.quick.common.api.client.mirror;
 
 import com.bakdata.quick.common.config.MirrorConfig;
 import java.util.Objects;
 import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
 import okhttp3.HttpUrl;
-import okhttp3.HttpUrl.Builder;
 
 /**
  * Utility for setting a Mirror host in Quick.
  */
 @Getter
 @Slf4j
-public class MirrorHost {
+public final class MirrorHost {
     private static final String DEFAULT_MIRROR_HOST_PATH = "mirror";
     private static final String DEFAULT_MIRROR_SCHEME = "http";
-    private final String topic;
-    private final MirrorConfig config;
-    private final String host;
+    private final HttpUrl url;
 
     /**
      * Private to creates the host with the topic name and mirror config. The host can be a service name or an IP.
      *
-     * @param topic the topic of the mirror.
-     * @param config mirror config to use. This can set the service prefix and REST path.
+     * @param url the full URL to connect to a Mirror.
      */
-    public MirrorHost(final String topic, final MirrorConfig config) {
-        this.topic = topic;
-        this.config = config;
-        this.host = this.config.getPrefix() + this.topic;
+    private MirrorHost(final HttpUrl url) {
+        this.url = url;
+    }
+
+    /**
+     * Creates the mirror host with the mirrorName. It adds the default {@link MirrorConfig#DEFAULT_MIRROR_HOST_PREFIX}
+     * to the mirror name
+     *
+     * @param mirrorName the name of the Mirror.
+     */
+    public static MirrorHost createMirrorHostWithDefaultPrefix(final String mirrorName) {
```

Comment:
```suggestion
    public static MirrorHost createMirrorHostWithPrefix(final String mirrorName) {
```

Might not be the default if configured

Reply:
You could also drop the `MirrorHost` both here and `createMirrorHostWithNoPrefix`. 
`MirrorHost.createWith(No)Prefix` sounds good to me.